warp10/src/main/java/io/warp10/script/functions/TIMEBOX.java

```diff
@@ -35,44 +35,47 @@
 import io.warp10.script.WarpScriptStackFunction;
 
 public class TIMEBOX extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
+
   /**
    * Default timeboxing is 30s
    */
   private static final long DEFAULT_TIMEBOX_MAXTIME = 30000L;
-  
+
   /**
    * Maximum timeboxing possible
    */
   private static final long TIMEBOX_MAXTIME;
-  
+
   static {
     TIMEBOX_MAXTIME = Long.parseLong(WarpConfig.getProperty(Configuration.CONFIG_WARPSCRIPT_TIMEBOX_MAXTIME, Long.toString(DEFAULT_TIMEBOX_MAXTIME)));
   }
-  
+
   public TIMEBOX(String name) {
     super(name);
   }
-  
+
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object top = stack.pop();
-    
+
     if (!(top instanceof Long)) {
       throw new WarpScriptException(getName() + " expects a maximum execution time on top of the stack.");
     }
-    
+
     long maxtime = Math.min(Math.max(0L,((Number) top).longValue()/Constants.TIME_UNITS_PER_MS), TIMEBOX_MAXTIME);
-    
+
     top = stack.pop();
-    
+
     if (!(top instanceof Macro)) {
       throw new WarpScriptException(getName() + " operates on a macro.");
     }
 
     final Macro macro = (Macro) top;
     final WarpScriptStack fstack = stack;
-    
+
+    Object timeboxed = Boolean.TRUE.equals(fstack.getAttribute(WarpScriptStack.ATTRIBUTE_TIMEBOXED));
```

Comment:
`Boolean timeboxed` would be more readable.

Reply:
If that makes any sense to store this information, it might be preferable to store a Long with maxtime value.